util/annotations/annotations.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package annotations
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Annotations is a general wrapper for text that is shown in the
+// Prometheus/Grafana UI along with the query results.
+type Annotations map[string]error
+
+func (a *Annotations) AddAnnotation(err error) {
+	(*a)[err.Error()] = err
+}
+
+func InitAnnotations(err error) Annotations {
```

Comment:
I would call this `New` and without the `error` parameter (creating an empty `Annotations`). Combined with `Add` above, you can then simply write `a := annotations.New().Add(err)`

With the proposed safeguard for `Add` (see above), you could also not have this method at all and write `annotations.Annotations{}.Add(err)`, but that has stutter again, so maybe a `New` function is better.